### PR TITLE
MCOL-4025 systemd units now preload libjemalloc

### PR DESCRIPTION
**Pull request by jmrojas2332**
ExeMgr, PrimProc, and WriteEngineServer now preload libjemalloc.

Also added MariaDB-server package as a dependency for the engine.

### CODE REVIEW

File: oam/install_scripts/columnstore-post-install.in [hunk 1]
```diff
@@ -177,13 +177,17 @@ if [ -z "aws" ]; then
...
 postConfigure
-systemctl start mariadb-columnstore
 
-# Wait for all columnstore to be ready, DDLProc is final process in startup order
-while  [ -z  "$(pgrep -x DDLProc)" ];
-do
-    sleep 1
-done
+systemctl cat mariadb-columnstore.service > /dev/null 2>&1
+if [ $? -eq 0 ] && [ $(running_systemd) -eq 0 ]; then
```
Comment (jmrojas2332) [inline oam/install_scripts/columnstore-post-install.in:182]: We shouldn't be starting a non-existent systemd unit so let's include this check.

In the future we may have to add an else to take into account what we want to happen on Sky, but for tomorrow's release this is sufficient.

File: oam/install_scripts/mcs-exemgr.service.in [hunk 2]
```diff
@@ -5,7 +5,7 @@ After=mcs-writeengineserver.service
 
 [Service]
 Type=simple
-Environment="SKIP_OAM_INIT=1"
+Environment="LD_PRELOAD=libjemalloc.so.1 libjemalloc.so.2"
```
Comment (jmrojas2332) [inline oam/install_scripts/mcs-exemgr.service.in]: There is good news; there is bad news. LD_PRELOAD will look for shared libs in the standard lib paths (i.e., /usr/lib64, /usr/lib, /usr/lib/x86_64-linux-gnu) so we do not actually need a direct path to libjemalloc.so. The issue we have is we won't be sure which version (so.1, so.2) user will have until runtime. Setting this environment variable requires an exact name, and not a dynamic variable, so the quick solution is to cut our loss (the one non-terminating error msg in journalctl for the lib that doesn't exist). I had a separate thought, where we determine which version of libjemalloc we have during cmake, however we cannot be certain that user has libjemalloc installed at that time, which can introduce the case where LD_PRELOAD="" and then we never load libjemalloc.

File: oam/install_scripts/mcs-primproc.service.in [hunk 3]
```diff
@@ -6,7 +6,7 @@ After=mcs-controllernode.service
 
 [Service]
 Type=simple
-Environment="SKIP_OAM_INIT=1"
+Environment="LD_PRELOAD=libjemalloc.so.1 libjemalloc.so.2"
```
Comment (jmrojas2332) [inline oam/install_scripts/mcs-primproc.service.in]: Same thing here as mcs-exemgr

File: oam/install_scripts/mcs-writeengineserver.service.in [hunk 4]
```diff
@@ -5,7 +5,7 @@ After=mcs-primproc.service
 
 [Service]
 Type=simple
-Environment="SKIP_OAM_INIT=1"
+Environment="LD_PRELOAD=libjemalloc.so.1 libjemalloc.so.2"
```
Comment (jmrojas2332) [inline oam/install_scripts/mcs-writeengineserver.service.in]: Same thing here as mcs-exemgr

File: oam/install_scripts/post-mysql-install [hunk 5]
```diff
@@ -22,10 +22,7 @@ checkForError() {
         # See if engine columnstore exist
         #---------------------------------------------------------------------------
         echo "checking for engine columnstore..."
-        mysql \
-                --user=root \
-                --execute='show engines;' \
-                | grep -i columnstore
+        su -s /bin/sh -c 'mysql --execute="show engines"' mysql 2> ${tmpdir}/post-mysql-install.log | grep -i columnstore
```
Comment (jmrojas2332) [inline oam/install_scripts/post-mysql-install:25]: should be running this using su.

File: oam/install_scripts/mcs-exemgr.service.in (hunk 2 above)
Comment (pleblanc1976) [reply oam/install_scripts/mcs-exemgr.service.in]: can we say that regular libjemalloc.so (no version) always exists?  I think we're going to always get an error msg from these, b/c at least one of them won't exist.

Comment (pleblanc1976) [reply oam/install_scripts/mcs-exemgr.service.in]: no, looks like the non-versioned name doesn't exist on deb systems, at least not my mint 19 box.  pita.  Might have to get fancy and find which one to use at runtime.  For reference, I had to add code like that to the old 'columnstore' script.

Comment (jmrojas2332) [reply oam/install_scripts/mcs-exemgr.service.in]: I have modified this to not get an error and always choose correct .so

File: oam/install_scripts/mcs-controllernode.service.in [hunk 6]
```diff
@@ -5,6 +5,7 @@ After=mcs-workernode.service
 
 [Service]
 Type=forking
+ExecStartPre=/usr/bin/env bash -c "systemctl start mcs-workernode"
```
Comment (jmrojas2332) [inline oam/install_scripts/mcs-controllernode.service.in:8]: Fixes mcs-columnstore restart not starting mcs-workernode again. Let's ensure mcs-workernode is active

File: oam/install_scripts/mcs-ddlproc.service.in [hunk 7]
```diff
@@ -1,6 +1,6 @@
 [Unit]
 Description=mcs-ddlproc
-PartOf=mcs-exemgr.service
+PartOf=mcs-writeengineserver.service
```
Comment (jmrojas2332) [inline oam/install_scripts/mcs-ddlproc.service.in:3]: Modified and tested ordering change to remove a bug and error/warning from mcs-exemgr, mcs-ddlproc

Ordering is primproc, exemgr, writeenginserver

File: oam/install_scripts/mcs-exemgr.service.in [hunk 8]
```diff
@@ -1,12 +1,11 @@
 [Unit]
 Description=mcs-exemgr
-PartOf=mcs-writeengineserver.service
-After=mcs-writeengineserver.service
+PartOf=mcs-primproc.service
```
Comment (jmrojas2332) [inline oam/install_scripts/mcs-exemgr.service.in:3]: Modified and tested ordering change to remove a bug and error/warning from mcs-exemgr, mcs-ddlproc

Ordering is primproc, exemgr, writeenginserver

File: oam/install_scripts/mcs-exemgr.service.in [hunk 9]
```diff
@@ -1,12 +1,11 @@
...
+After=mcs-primproc.service
 
 [Service]
 Type=simple
-Environment="LD_PRELOAD=libjemalloc.so.1 libjemalloc.so.2"
-ExecStart=@ENGINE_BINDIR@/ExeMgr
+ExecStart=/usr/bin/env bash -c "LD_PRELOAD=$(ldconfig -p | grep -m1 libjemalloc | awk '{print $1}') exec @ENGINE_BINDIR@/ExeMgr"
```
Comment (jmrojas2332) [inline oam/install_scripts/mcs-exemgr.service.in:8]: `$(ldconfig -p | grep -m1 libjemalloc | awk '{print $1}')` brings in the magic to get us correct .so (e.g., libjemalloc.so.1, libjemalloc.so.2). 
Ensured libjemalloc was loaded by looking at  `lsof -p $(pgrep ExeMgr)`

Note: LD_PRELOAD will look for shared libs in the standard lib paths (i.e., /usr/lib64, /usr/lib, /usr/lib/x86_64-linux-gnu) so we do not actually need a direct path to libjemalloc.so

File: oam/install_scripts/mcs-primproc.service.in [hunk 10]
```diff
@@ -6,8 +6,8 @@ After=mcs-controllernode.service
 
 [Service]
 Type=simple
-Environment="LD_PRELOAD=libjemalloc.so.1 libjemalloc.so.2"
-ExecStart=@ENGINE_BINDIR@/PrimProc
+ExecStart=/usr/bin/env bash -c "LD_PRELOAD=$(ldconfig -p | grep -m1 libjemalloc | awk '{print $1}') exec @ENGINE_BINDIR@/PrimProc"
+ExecStartPost=sleep 2
```
Comment (jmrojas2332) [inline oam/install_scripts/mcs-primproc.service.in:10]: Prevents "Could not connect to PMS1: Connection refused" errors in exemgr and ddlprocs journal

**Review by pleblanc1976 — APPROVED** [review]

File: oam/install_scripts/columnstore-post-install.in [hunk 11]
```diff
@@ -177,13 +177,17 @@ if [ -z "aws" ]; then
...
+    systemctl start mariadb-columnstore
```
Comment (drrtuy) [inline oam/install_scripts/columnstore-post-install.in:183]: We might use syncrounous mcs-* units to avoid the loop.

**Pull request merged**